src/x86/sse41.rs

```diff
@@ -238,11 +239,42 @@ pub unsafe fn _mm_max_epi32(a: i32x4, b: i32x4) -> i32x4 {
 /// maximum values.
 #[inline(always)]
 #[target_feature = "+sse4.1"]
-#[cfg_attr(test, assert_instr(pmaxud, imm8 = 0))]
+#[cfg_attr(test, assert_instr(pmaxud))]
 pub unsafe fn _mm_max_epu32(a: u32x4, b: u32x4) -> u32x4 {
     pmaxud(a, b)
 }
 
+/// Convert packed 32-bit integers from `a` and `b` to packed 16-bit integers using unsigned saturation
+#[inline(always)]
+#[target_feature = "+sse4.1"]
+#[cfg_attr(test, assert_instr(packusdw))]
+pub unsafe fn _mm_packus_epi32(a: i32x4, b: i32x4) -> u16x8 {
+    packusdw(a, b)
+}
+
+/// Compare packed 64-bit integers in `a` and `b` for equality
+#[inline(always)]
+#[target_feature = "+sse4.1"]
+#[cfg_attr(test, assert_instr(pcmpeqq))]
+pub unsafe fn _mm_cmpeq_epi64(a: i64x2, b: i64x2) -> i64x2 {
+    a.eq(b)
+}
+
+/// Sign extend packed 8-bit integers in `a` to packed 16-bit integers
+#[target_feature = "+sse4.1"]
+#[cfg_attr(test, assert_instr(pmovsxbw))]
+pub unsafe fn _mm_cvtepi8_epi16(a: i8x16) -> i16x8 {
+    simd_cast::<::v64::i8x8, _>(simd_shuffle8(a, a, [0, 1, 2, 3, 4, 5, 6, 7]))
```

Comment:
I think you can say `.as_i16x8()` here?

Reply:
Yes, but the type still needs to be specified, for example:

```rust
simd_shuffle8::<_, ::v64::i8x8>(a, a, [0, 1, 2, 3, 4, 5, 6, 7]).as_i16x8()
```

Reply:
Yeah I'd prefer that. I'd like to use the wrappers around the `simd` intrinsics as much as possible.

Reply:
Done ;)